lib/elasticsearch.js

```diff
@@ -0,0 +1,593 @@
+'use strict';
+
+const AWS = require('aws-sdk');
+const AWSESConnection = require('http-aws-es');
+
+const { URL } = require('url');
+
+const UUID = require('uuid/v4');
+
+const elasticsearch = require('@elastic/elasticsearch');
+
+const ElasticSearchFilters = require('./elasticsearch-filters');
+const ElasticSearchError = require('./elasticsearch-error');
+
+const DEFAULT_LIMIT = 500;
+const ES_DEFAULT_HOST = 'http://localhost';
+
+/**
+ * @class ElasticSearch
+ * @classdesc Elasticsearch driver module
+*/
+class ElasticSearch {
+
+	constructor(config = {}) {
+		if(typeof config !== 'object' || Array.isArray(config))
+			throw new ElasticSearchError('Invalid config', ElasticSearchError.codes.INVALID_CONFIG);
+
+		this.config = {
+			protocol: config.protocol || '',
+			host: config.host || ES_DEFAULT_HOST,
+			port: config.port || '',
+			user: config.user || '',
+			password: config.password || '',
+			limit: config.limit || DEFAULT_LIMIT,
+			awsCredentials: config.awsCredentials || false
+		};
+
+		try {
+			const hostProtocol = `${new URL(this.config.host).protocol}//`; // throws if the URL doesn't have a protocol
+			this.config.protocol = hostProtocol;
+		} catch(err) {
+			// do nothing...
+		}
+
+		this.config.host = this.config.host.replace(this.config.protocol, ''); // Avoid protocol duplication
```

Comment:
Ya que vas a usar `URL` para detectar el protocolo. Podrías usarlo para que el `protocol` no se repita en el `host`
```suggestion
		try {
			const url = new URL(this.config.host); // throws if the URL doesn't have a protocol
			this.config.protocol = `${url.protocol}//`;
			this.config.host = url.host;
		} catch(err) {
			// do nothing...
		}
```